stl/inc/utility

```diff
@@ -80,14 +110,29 @@ _CONSTEXPR20 void iter_swap(_FwdIt1 _Left, _FwdIt2 _Right) { // swap *_Left and
 
 template <class _Ty, size_t _Size, enable_if_t<_Is_swappable<_Ty>::value, int> _Enabled>
 _CONSTEXPR20 void swap(_Ty (&_Left)[_Size], _Ty (&_Right)[_Size]) noexcept(_Is_nothrow_swappable<_Ty>::value) {
-    if (&_Left != &_Right) {
-        _Ty* _First1 = _Left;
-        _Ty* _Last1  = _First1 + _Size;
-        _Ty* _First2 = _Right;
-        for (; _First1 != _Last1; ++_First1, ++_First2) {
-            _STD iter_swap(_First1, _First2);
+    if (&_Left == &_Right) {
+        return;
+    }
+
+    _Ty* _First1 = _Left;
+    _Ty* _Last1  = _First1 + _Size;
+    _Ty* _First2 = _Right;
+
+#if _USE_STD_VECTOR_ALGORITHMS
+    if constexpr (_Is_trivially_swappable_v<_Ty>) {
+#if _HAS_CXX20
+        if (!_STD is_constant_evaluated())
+#endif // _HAS_CXX20
+        {
+            __std_swap_ranges_trivially_swappable_noalias(_First1, _Last1, _First2);
+            return;
         }
     }
+#endif // _USE_STD_VECTOR_ALGORITHMS
+
+    for (; _First1 != _Last1; ++_First1, ++_First2) {
+        _STD iter_swap(_First1, _First2);

```

Comment:
Pre-existing: `iter_swap` may as well be inlined here. The Standard (now) forbids program-defined specialization of `iter_swap`, and even if it didn't specializing for `my_trivial_type*` would be super weird.

Reply:
I don't see the Standardese for that. I would prefer not to make such a drive-by change in this PR - could you file a tracking issue? It seems that it would be better to perform a global cleanup (presumably with comments about how we're intentionally inlining `iter_swap`, and calling `swap` with ADL).

Reply:
I've filed #2692 "STL:: `std::iter_swap` can be inlined into callers".